### DISCUSSION TITLE
ADS1256 - PI4J-v2

### DISCUSSION THREAD

**Discussion by weesftw** [Pi4J/pi4j#229] [Q&A] [answered]
Guys, could someone help me to work with **ADS1256** with the PI4J-v2 lib? With examples of connections on SPI ports.

If you have a similar project, I would be grateful.

**Comment by taartspi** [comment]
I implemented a MCP3008, 8 bit ADC SPI connected.     See https://github.com/Pi4J/pi4j-device-tca9548    

  MCP3008 A/D 10bit converter
src/main/java/com/pi4j/devices/mcp3008/README.md
This package does not use any other code within my repo.

 Tom

**Reply by TutsIR** [reply]
Thanks Tom. This was the only working example I could find for mcp3008.

**Comment by weesftw** [comment]
Hey @taartspi thanks for the reply.

Could you help me create an example to work with `ADS1256`? It's a fact, I'm having great difficulties even looking at the source code of `MCP3008` that you sent me.

**Reply by taartspi** [reply]
When I read this I looked at a datasheet for the ADS1256 and saw it  too complex to try and create a new SPI class without having the HW for testing.  So I ordered a chip and it will arrive late this week.  Once I have the chip I will look into coding a class so we have something in front of us to refer to during debug.
Also, you mentioned visibility to my MCP3008,  the repository is public.
https://github.com/Pi4J/pi4j-device-tca9548    From here you should be able to follow src/main/... to the device.  Let me know if you cannot use this link.
Tom

**Reply by weesftw** [reply]
@taartspi 

Tom, thanks for the feedback. I was able to locate and access the project that is using the `MCP3008` chip. I was using it as a template to try to code a class for `ADS1256` but without success.

I'm so glad you can help me with this challenge, I'm so excited.

I look forward to more updates, and again, thank you very much.

**Comment by taartspi** [comment]
Evening   

I ordered an ADS1256 from waveform and see on their website it has finally
left China for the US.  Hopefully not on a ship.

I coded what I intend to unit test and placed it in that same repo.   It
has NO testing, but based on the waveform details in their C program.  When
the chip arrives I will unit test and see if I can get it working.   Just
wanted  to let you know this and not appear as though I forgot.     If you
were able to have success on your end let me see what you coded.

take care  Tom

**Reply by weesftw** [reply]
Greetings @taartspi, thanks for the feedback.

I've been following the commits of the `pi4j-device-tca9548` repository, you don't know how happy I am for the dedication you had to fulfill my request, this will help me a lot to better understand the chipset and create several experiments.

 I look forward to the next chapters!

See you soon.

**Comment by taartspi** [comment]
I did get one ADS1256 chip.  I just pushed code that works with the chip.  Look at the README to see what pins this chip exposed.  When I get the Waveshare module I will test with it as that chip exposes all the lines from the ADS1256.

But, I hit a problem in the Pi4J code base. It was a simple fix and I did the push/pull activity and will wait to see if the Pi4J team agrees with my change.  You could setup the environment to build the Pi4j base and easily make this change on your user level.  I use maven native on a pi-4 8 gig so making a change to the Pi4j base is easy, I do not know what build environment you use.    Hopefully there is a quick response and the team agrees on the fix and you can build it yourself and know it will be generally available in the future.

**Comment by taartspi** [comment]
I completed testing with ab AD/DA Board from WaveShare.  This ADS1256 exposes all the GPIO lines.  I think this is what you asked for,  if you want any assistance to build that MODE1 Pi4j fix let me know. I believe week after next the Pi4J team will be available to see how they would release the MODE1 fix.

When I have time I will implement using the four DIOs on the ADS1256

**Comment by weesftw** [comment]
@taartspi

Tom, thank you so much for all your help. Between this week and the weekend I'm going to get the coded class and do some tests on them with the board, I really appreciate your help and if I have any problems I'll contact you.

**Reply by taartspi** [reply]
#235    #236 
I believe one of those are the link to the discussion on mode1 failing and the other my pull request with a simple fix for that provider problem

And you are welcome

**Comment by taartspi** [comment, marked as answer]
The fix for MODE_1 SPI has been incorporated into the 2.2.0-SNAPSHOT.  So refresh the jar files and you can use that example ads1256 program to  begin your work.     If there is confusion to change what I did for your specific needs let me know.  Tom

**Reply by FDelporte** [reply]
Even better, it's in the release version ;-)
So you can update your dependency to 2.2.0

See https://pi4j.com/about/release-notes/
Code changes: https://github.com/Pi4J/pi4j-v2/compare/2.1.1...2.2.0
